exporter/kineticaexporter/config.go

```diff
@@ -0,0 +1,253 @@
+
+	if !fileExists(cfg.LogConfigFile) {
+		fmt.Println("WARNING : LOG config file ", cfg.LogConfigFile, " does not exist; will work with default logger ...")
+	}
+
+	fmt.Println("Password = ", string(cfg.Password))
+
+	return nil
+}
+
+func parseNumber(s string, fallback int) int {
+	v, err := strconv.Atoi(s)
+	if err == nil {
+		return v
+	}
+	return fallback
+}
+
+// createLogger - creates a Logger using user defined configuration from a file.
+// If the file name is not found or there is some error it will create a default
+// logger using the default logger config file named 'config_log_zap.yaml'.
+//
+//	@receiver cfg
+//	@return *zap.Logger
+func (cfg *Config) createLogger() *zap.Logger {
+
+	if kineticaLogger != nil {
+		return kineticaLogger
+	}
+
+	var logConfig zap.Config
+
+	yamlFile, _ := os.ReadFile(cfg.LogConfigFile)
+	if err := yaml.Unmarshal(yamlFile, &logConfig); err != nil {
+		fmt.Println("Error in loading log config file : ", cfg.LogConfigFile)
+		fmt.Println(err)
+		fmt.Println("Creating default logger ...")
+		kineticaLogger = cfg.createDefaultLogger()
+		return kineticaLogger
+	}
+	fmt.Println("Log Config from YAML file : ", logConfig)
+
+	var (
+		stdout      zapcore.WriteSyncer
+		file        zapcore.WriteSyncer
+		logFilePath *url.URL
+	)
+
+	for _, path := range logConfig.OutputPaths {
+
+		switch {
+		case path == "stdout" || path == "stderror":
+			stdout = zapcore.AddSync(os.Stdout)
+			fmt.Println("Created stdout syncer ...")
+		case strings.HasPrefix(path, "lumberjack://"):
+			var err error
+			logFilePath, err = url.Parse(path)
+			fmt.Println("LogFilePath : ", logFilePath)
+
+			if err == nil {
+				filename := strings.TrimLeft(logFilePath.Path, "/")
+				fmt.Println("LogFileName : ", filename)
+
+				if filename != "" {
+					q := logFilePath.Query()
+					l := &lumberjack.Logger{
+						Filename:   filename,
+						MaxSize:    parseNumber(q.Get("maxSize"), maxSize),
+						MaxAge:     parseNumber(q.Get("maxAge"), maxAge),
+						MaxBackups: parseNumber(q.Get("maxBackups"), maxBackups),
+						LocalTime:  false,
+						Compress:   false,
+					}
+
+					file = zapcore.AddSync(l)
+					fmt.Println("Created file syncer ...")
+				}
+			}
+		default:
+			// Unknown output format
+			fmt.Println("Invalid output path specified in config ...")
+		}
+	}
+
+	if stdout == nil && file == nil {
+		fmt.Println("Both stdout and file not available; creating default logger ...")
+		return cfg.createDefaultLogger()
+	}
+
+	level := zap.NewAtomicLevelAt(logConfig.Level.Level())
+
+	consoleEncoder := zapcore.NewConsoleEncoder(logConfig.EncoderConfig)
+	fileEncoder := zapcore.NewJSONEncoder(logConfig.EncoderConfig)
+
+	core := zapcore.NewTee(
+		zapcore.NewCore(consoleEncoder, stdout, level),
+		zapcore.NewCore(fileEncoder, file, level),
+	)
+
+	kineticaLogger = zap.New(core)
+	return kineticaLogger
+}
+
+// createDefaultLogger
+//
+//	@receiver cfg
+//	@return *zap.Logger
+func (cfg *Config) createDefaultLogger() *zap.Logger {
```

Comment:
Is this part of the exporting logic? If it is creating a logger to use within the component but not as part of the data exporting then we should be using the logger that the collector provides instead. If it is exporting logic please remove it from this PR and we'll add it in a future PR.

Reply:
Removed this method since this was intended to be specific to our internal logging requirements.